language/vm/src/file_format.rs

```diff
@@ -676,7 +680,7 @@ impl SignatureToken {
     pub fn is_primitive(&self) -> bool {
```

Comment:
It looks like the only place this function is used is in checking for valid tx script signatures:

```
/// This function checks the extra requirements on the signature of the main function of a script.
pub fn verify_main_signature(script: &CompiledScript) -> VMResult<()> {
    let arguments = script.signature_at(script.as_inner().parameters);
    for arg_type in &arguments.0 {
        if !(arg_type.is_primitive()
            || *arg_type == SignatureToken::Vector(Box::new(SignatureToken::U8)))
        {
            return Err(VMStatus::new(StatusCode::INVALID_MAIN_FUNCTION_SIGNATURE));
        }
    }
    Ok(())
}
```

If this is the only use-case, it might make sense for us to repurpose this to `is_valid_tx_script_parameter` or similar?  Eventually, we want to allow tx scripts to accept `vector<u64>` etc. and `&signer`, but presumably not `signer`.

Reply:
even better would be to remove it completely and add that to `verify_main_signature` which as mentioned will change significantly

Reply:
I'll take a look in just killing it